db/compaction_picker.cc

```diff
@@ -297,8 +297,13 @@ Compaction* CompactionPicker::CompactFiles(
   // shouldn't have been released since.
   assert(!FilesRangeOverlapWithCompaction(input_files, output_level));
 
+  auto ioptions = ioptions_;
```

Comment:
hmm the `Compaction` object stores a reference to these options, and I think they can go out of scope before the `Compaction` object is destroyed. should we just pass max_subcompactions in as a separate parameter?

Reply:
Actually, the `Compaction` object stores a copy of that.
But I think passing `max_subcompactions` as a parameter is more clear, I'll do that later.

Reply:
oh right, sorry I misread the code.